### ISSUE TITLE
Duplicate records found

### ISSUE THREAD

**Issue by wdbasson** [didx-xyz/aries-cloudapi-python#477]
- [ ] test against k8s in dev
- [ ] promote to test env

```
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: 2023-09-28 09:56:24,765 aries_cloudagent.admin.server ERROR Handler error with exception: Duplicate records found
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: Traceback (most recent call last):
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 173, in ready_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 210, in debug_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aiohttp_apispec/middlewares.py", line 45, in validation_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 330, in check_token
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 379, in check_multitenant_authorization
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 442, in setup_context
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await task
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/futures.py", line 284, in __await__
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     yield self  # This tells Task to wait for completion.
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/tasks.py", line 328, in __wakeup
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     future.result()
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/futures.py", line 201, in result
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     raise self._exception
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     result = coro.send(None)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/protocols/endorse_transaction/v1_0/routes.py", line 601, in set_endorser_role
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     jobs = await record.metadata_get(session, "transaction_jobs")
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/connections/models/conn_record.py", line 550, in metadata_get
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     record = await storage.find_record(
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/storage/askar.py", line 167, in find_record
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     raise StorageDuplicateError("Duplicate records found")
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: aries_cloudagent.storage.error.StorageDuplicateError: Duplicate records found
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: 2023-09-28 09:56:24,773 aiohttp.server ERROR Error handling request
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: Traceback (most recent call last):
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     resp = await request_handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     resp = await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aiohttp/web_middlewares.py", line 117, in impl
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 173, in ready_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 210, in debug_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aiohttp_apispec/middlewares.py", line 45, in validation_middleware
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 330, in check_token
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 379, in check_multitenant_authorization
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await handler(request)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/admin/server.py", line 442, in setup_context
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     return await task
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/futures.py", line 284, in __await__
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     yield self  # This tells Task to wait for completion.
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/tasks.py", line 328, in __wakeup
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     future.result()
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/futures.py", line 201, in result
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     raise self._exception
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/usr/local/lib/python3.9/asyncio/tasks.py", line 256, in __step
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     result = coro.send(None)
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/protocols/endorse_transaction/v1_0/routes.py", line 601, in set_endorser_role
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     jobs = await record.metadata_get(session, "transaction_jobs")
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/connections/models/conn_record.py", line 550, in metadata_get
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     record = await storage.find_record(
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:   File "/home/aries/.local/lib/python3.9/site-packages/aries_cloudagent/storage/askar.py", line 167, in find_record
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]:     raise StorageDuplicateError("Duplicate records found")
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: aries_cloudagent.storage.error.StorageDuplicateError: Duplicate records found
dev-cloudapi/governance-multitenant-agent-7fdd5666d6-d8bml[governance-multitenant-agent]: 2023-09-28 09:56:24,778 aiohttp.access INFO 10.1.97.59 [28/Sep/2023:09:56:24 +0000] "POST /transactions/639d3e8c-e111-4129-ae31-0b2f0a6eac47/set-endorser-role?transaction_my_job=TRANSACTION_AUTHOR HTTP/1.1" 500 244 "-" "Python/3.9 aiohttp/3.8.5"
```

**Comment by ff137** [comment]
Will be resolved in #492

**Comment by wdbasson** [comment]
Original sleep statement that identified where the race condition was in the code before refactoring of onboarding process:

`app/services/onboarding.py`
```
    async def set_endorser_roles(endorser_connection, connection_record):
        bound_logger.debug("Setting roles for endorser")
        await endorser_controller.endorse_transaction.set_endorser_role(
            conn_id=endorser_connection["connection_id"],
            transaction_my_job="TRANSACTION_ENDORSER",
        )

        bound_logger.debug("XXX Set roles for endorser...sleep 5")
        time.sleep(5)
        await issuer_controller.endorse_transaction.set_endorser_role(
            conn_id=connection_record.connection_id,
            transaction_my_job="TRANSACTION_AUTHOR",
        )
        bound_logger.debug("Successfully set roles for endorser.")

    async def configure_endorsement(connection_record, endorser_did):
        # Make sure endorsement has been configured
        # There is currently no way to retrieve endorser info. We'll just set it
        # to make sure the endorser info is set.
        bound_logger.debug("Setting endorser info")
        await issuer_controller.endorse_transaction.set_endorser_info(
            conn_id=connection_record.connection_id,
            endorser_did=endorser_did.did,
        )
        bound_logger.debug("Successfully set endorser info.")
```
